src/Microsoft.OData.Core/Evaluation/LiteralFormatter.cs

```diff
@@ -223,6 +224,11 @@ private static string FormatRawLiteral(object value)
                 return value.ToString();
             }
 
+            if (value is DateOnly dateOnly)
+            {
+                return ((Date)dateOnly).ToString();

```

Comment:
I wonder whether it's necessary to cast to `Date` here? Could we consider creating Date-to-string helper function that could be used to ensure consistency for both `Date` and `DateOnly` but could allow us to work with `DateOnly` without having to cast to `Date`? I think if `DateOnly` is used often or the preferred choice in the future, we should consider having first-class support for it and avoid the overhead of casting to `Date` when possible.

Reply:
In the next breaking change version (9 version), we should remove 'Date' 'TimeOfDay' and use 'DateOnly' and 'TimeOnly' types introduced in .NET. 
In current 8 version, I try to avoid breaking change. And most important, we should make sure to get the same 'literal' string no matter whether customers are using 'Date' or 'DateOnly'.  

DateOnly has several 'ToString()' overloads, the string literal varys along with the formatter provider. 'Cast' to Date then call 'ToString' on 'Date' is a 'center' method and make sure we get the same literal string no matter what type is. And to call 'ToString()' is a straightforward. There's no need to create a helper method to wrapper 'ToString()' again. What do you think?

Reply:
I was curious about the potential overhead of casting to `Edm.Date` so I ran some tests to measure the `ToString()` scenario. To my surprise, casting to `Edm.Date` then call `ToString()` (i.e. `((Date)dateOnly).ToString()` appears to be faster than `dateOnly.ToString("yyyy-MM-dd", InvariantCulture)`.

This issue appears to be specific case of `DateOnly.ToString(format, provider)` overload which [verifies that format is a valid date-only format](https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/DateOnly.cs,772), and this verification happens in O(n). You can bypass this overhead by converting the `DateOnly` to `DateTime` then calling `DateTime.ToString("yyyy-MM-dd", InvariantCulture)`. This happens to be faster than casting to `Edm.Date`.

![image](https://github.com/user-attachments/assets/4b3cf333-ee37-48da-a7fa-530c3ef05d3b)



Here's the [benchmark code](https://sharpbench.dev/#s:gqF2AaFj2iFOLy8gdmlzaXQgaHR0cHM6Ly9iZW5jaG1hcmtkb3RuZXQub3JnLyBmb3IgbW9yZSBpbmZvIG9uIEJlbmNobWFya0RvdE5ldAoKdXNpbmcgQmVuY2htYXJrRG90TmV0LkF0dHJpYnV0ZXM7CnVzaW5nIEJlbmNobWFya0RvdE5ldC5EaWFnbm9zZXJzOwp1c2luZyBTeXN0ZW0uR2xvYmFsaXphdGlvbjsKdXNpbmcgRWRtID0gTWljcm9zb2Z0Lk9EYXRhLkVkbTsKCltDYXRlZ29yaWVzQ29sdW1uXQpwdWJsaWMgY2xhc3MgQmVuY2htYXJrcwp7CiAgICBwcml2YXRlIHN0YXRpYyByZWFkb25seSBFZG0uRGF0ZSBlZG1EYXRlID0gbmV3IEVkbS5EYXRlKDIwMjQsIDEsIDEpOwogICAgcHJpdmF0ZSBzdGF0aWMgcmVhZG9ubHkgRGF0ZU9ubHkgZGF0ZU9ubHkgPSBuZXcgRGF0ZU9ubHkoMjAyNCwgMSwgMSk7CiAgICBwcml2YXRlIHN0YXRpYyByZWFkb25seSBEYXRlVGltZSBkYXRlVGltZSA9IG5ldyBEYXRlVGltZSgyMDI0LCAxLCAxKTsKCiAgICBbQmVuY2htYXJrXQogICAgcHVibGljIHN0cmluZyBEYXRlVGltZVRvU3RyaW5nX0Jhc2VsaW5lKCkKICAgIHsKICAgICAgICByZXR1cm4gZGF0ZVRpbWUuVG9TdHJpbmcoInl5eXktTU0tZGQiLCBDdWx0dXJlSW5mby5JbnZhcmlhbnRDdWx0dXJlKTsKICAgIH0KCiAgICBbQmVuY2htYXJrXQogICAgcHVibGljIHN0cmluZyBFZG1EYXRlVG9TdHJpbmcoKQogICAgewogICAgICAgIHJldHVybiBlZG1EYXRlLlRvU3RyaW5nKCk7CiAgICB9CgogICAgW0JlbmNobWFya10KICAgIHB1YmxpYyBzdHJpbmcgRGF0ZU9ubHlUb1N0cmluZ19DYXN0VG9FZG1EYXRlKCkKICAgIHsKICAgICAgICByZXR1cm4gKChFZG0uRGF0ZSlkYXRlT25seSkuVG9TdHJpbmcoKTsKICAgIH0KCiAgICBbQmVuY2htYXJrXQogICAgcHVibGljIHN0cmluZyBEYXRlT25seVRvU3RyaW5nX0NvbnZlclRvRGF0ZVRpbWUoKQogICAgewogICAgICAgIHZhciBkYXRlVGltZSA9IG5ldyBEYXRlVGltZShkYXRlT25seSwgZGVmYXVsdCk7CiAgICAgICAgcmV0dXJuIChkYXRlVGltZSkuVG9TdHJpbmcoInl5eXktTU0tZGQiLCBDdWx0dXJlSW5mby5JbnZhcmlhbnRDdWx0dXJlKTsKICAgIH0KCiAgICBbQmVuY2htYXJrXQogICAgcHVibGljIHN0cmluZyBEYXRlT25seVRvU3RyaW5nX1dpdGhEYXRlVGltZUZvcm1hdHRlcigpCiAgICB7CiAgICAgICAgdmFyIGRhdGVUaW1lID0gbmV3IERhdGVUaW1lKGRhdGVPbmx5LCBkZWZhdWx0KTsKICAgICAgICByZXR1cm4gZGF0ZU9ubHkuVG9TdHJpbmcoInl5eXktTU0tZGQiLCBDdWx0dXJlSW5mby5JbnZhcmlhbnRDdWx0dXJlKTsKICAgIH0KCiAgICBbQmVuY2htYXJrXQogICAgcHVibGljIHN0cmluZyBEYXRlT25seVRvU3RyaW5nX05vQ2FzdCgpCiAgICB7CiAgICAgICAgcmV0dXJuIGRhdGVPbmx5LlRvU3RyaW5nKCJ5eXl5LU1NLWRkIiwgQ3VsdHVyZUluZm8uSW52YXJpYW50Q3VsdHVyZSk7CiAgICB9Cn0KCgpuYW1lc3BhY2UgTWljcm9zb2Z0Lk9EYXRhLkVkbQp7CiAgICAvLy8gPHN1bW1hcnk+CiAgICAvLy8gRGF0ZSB0eXBlIGZvciBFZG0uRGF0ZQogICAgLy8vIDwvc3VtbWFyeT4KICAgIHB1YmxpYyBzdHJ1Y3QgRGF0ZSA6IElDb21wYXJhYmxlLCBJQ29tcGFyYWJsZTxEYXRlPiwgSUVxdWF0YWJsZTxEYXRlPgogICAgewogICAgICAgIC8vLyA8c3VtbWFyeT4KICAgICAgICAvLy8gTWluIHZhbHVlIG9mIDxzZWUgY3JlZj0iTWljcm9zb2Z0Lk9EYXRhLkVkbS5EYXRlIi8+CiAgICAgICAgLy8vIDwvc3VtbWFyeT4KICAgICAgICBwdWJsaWMgc3RhdGljIHJlYWRvbmx5IERhdGUgTWluVmFsdWUgPSBuZXcgRGF0ZSgxLCAxLCAxKTsKCiAgICAgICAgLy8vIDxzdW1tYXJ5PgogICAgICAgIC8vLyBNYXggdmFsdWUgb2YgPHNlZSBjcmVmPSJNaWNyb3NvZnQuT0RhdGEuRWRtLkRhdGUiLz4KICAgICAgICAvLy8gPC9zdW1tYXJ5PgogICAgICAgIHB1YmxpYyBzdGF0aWMgcmVhZG9ubHkgRGF0ZSBNYXhWYWx1ZSA9IG5ldyBEYXRlKDk5OTksIDEyLCAzMSk7CgogICAgICAgIC8vLyA8c3VtbWFyeT4KICAgICAgICAvLy8gSW50ZXJuYWwgdXNpbmcgU3lzdGVtLkRhdGVUaW1lCiAgICAgICAgLy8vIDwvc3VtbWFyeT4KICAgICAgICBwcml2YXRlIERhdGVUaW1lIGRhdGVUaW1lOwoKICAgICAgICAvLy8gPHN1bW1hcnk+CiAgICAgICAgLy8vIENvbnN0cnVjdG9yIG9mIDxzZWUgY3JlZj0iTWljcm9zb2Z0Lk9EYXRhLkVkbS5EYXRlIi8+CiAgICAgICAgLy8vIDwvc3VtbWFyeT4KICAgICAgICAvLy8gPHBhcmFtIG5hbWU9InllYXIiPlllYXIgdmFsdWUgb2YgZGF0ZTwvcGFyYW0+CiAgICAgICAgLy8vIDxwYXJhbSBuYW1lPSJtb250aCI+TW9udGggdmFsdWUgb2YgZGF0ZTwvcGFyYW0+CiAgICAgICAgLy8vIDxwYXJhbSBuYW1lPSJkYXkiPkRheSB2YWx1ZSBvZiBkYXRlPC9wYXJhbT4KICAgICAgICBwdWJsaWMgRGF0ZShpbnQgeWVhciwgaW50IG1vbnRoLCBpbnQgZGF5KQogICAgICAgICAgICA6IHRoaXMoKQogICAgICAgIHsKICAgICAgICAgICAgdHJ5CiAgICAgICAgICAgIHsKICAgICAgICAgICAgICAgIHRoaXMuZGF0ZVRpbWUgPSBuZXcgRGF0ZVRpbWUoeWVhciwgbW9udGgsIGRheSk7CiAgICAgICAgICAgIH0KICAgICAgICAgICAgY2F0Y2ggKFN5c3RlbS5Bcmd1bWVudE91dE9mUmFuZ2VFeGNlcHRpb24pCiAgICAgICAgICAgIHsKICAgICAgICAgICAgICAgIHRocm93IG5ldyBGb3JtYXRFeGNlcHRpb24oIkludmFsaWQgcGFyYW1ldGVycyIpOwogICAgICAgICAgICB9CiAgICAgICAgfQoKICAgICAgICAvLy8gPHN1bW1hcnk+CiAgICAgICAgLy8vIEdldHMgYSA8c2VlIGNyZWY9Ik1pY3Jvc29mdC5PRGF0YS5FZG0uRGF0ZSIvPiBvYmplY3QgdGhhdCBpcyBzZXQgdG8gY3VycmVudCBkYXRlIG9uIHRoaXMgY29tcHV0ZXIsIGV4cHJlc3NlZCBhcyB0aGUgbG9jYWwgdGltZS4KICAgICAgICAvLy8gPC9zdW1tYXJ5PgogICAgICAgIHB1YmxpYyBzdGF0aWMgRGF0ZSBOb3cKICAgICAgICB7CiAgICAgICAgICAgIGdldCB7IHJldHVybiBEYXRlVGltZS5Ob3c7IH0KICAgICAgICB9CgogICAgICAgIC8vLyA8c3VtbWFyeT4KICAgICAgICAvLy8gR2V0cyB0aGUgeWVhciBjb21wb25lbnQgb2YgdGhlIGRhdGUgcmVwcmVzZW50ZWQgYnkgdGhpcyBpbnN0YW5jZS4KICAgICAgICAvLy8gPC9zdW1tYXJ5PgogICAgICAgIHB1YmxpYyBpbnQgWWVhcgogICAgICAgIHsKICAgICAgICAgICAgZ2V0CiAgICAgICAgICAgIHsKICAgICAgICAgICAgICAgIHJldHVybiBkYXRlVGltZS5ZZWFyOwogICAgICAgICAgICB9CiAgICAgICAgfQoKICAgICAgICAvLy8gPHN1bW1hcnk+CiAgICAgICAgLy8vIEdldHMgdGhlIG1vbnRoIGNvbXBvbmVudCBvZiB0aGUgZGF0ZSByZXByZXNlbnRlZCBieSB0aGlzIGluc3RhbmNlLgogICAgICAgIC8vLyA8L3N1bW1hcnk+CiAgICAgICAgcHVibGljIGludCBNb250aAogICAgICAgIHsKICAgICAgICAgICAgZ2V0CiAgICAgICAgICAgIHsKICAgICAgICAgICAgICAgIHJldHVybiBkYXRlVGltZS5Nb250aDsKICAgICAgICAgICAgfQogICAgICAgIH0KCiAgICAgICAgLy8vIDxzdW1tYXJ5PgogICAgICAgIC8vLyBHZXRzIHRoZSBkYXkgb2YgdGhlIG1vbnRoIHJlcHJlc2VudGVkIGJ5IHRoaXMgaW5zdGFuY2UuCiAgICAgICAgLy8vIDwvc3VtbWFyeT4KICAgICAgICBwdWJsaWMgaW50IERheQogICAgICAgIHsKICAgICAgICAgICAgZ2V0CiAgICAgICAgICAgIHsKICAgICAgICAgICAgICAgIHJldHVybiBkYXRlVGltZS5EYXk7CiAgICAgICAgICAgIH0KICAgICAgICB9CgogICAgICAgIC8vLyA8c3VtbWFyeT4KICAgICAgICAvLy8gQ29udmVydCBEYXRlIHRvIENsciBEYXRlVGltZQogICAgICAgIC8vLyA8L3N1bW1hcnk+CiAgICAgICAgLy8vIDxwYXJhbSBuYW1lPSJvcGVyYW5kIj5EYXRlIFZhbHVlPC9wYXJhbT4KICAgICAgICAvLy8gPHJldHVybnM+RGF0ZVRpbWUgVmFsdWUgd2hpY2ggcmVwcmVzZW50IHRoZSBEYXRlPC9yZXR1cm5zPgogICAgICAgIHB1YmxpYyBzdGF0aWMgaW1wbGljaXQgb3BlcmF0b3IgRGF0ZVRpbWUoRGF0ZSBvcGVyYW5kKQogICAgICAgIHsKICAgICAgICAgICAgcmV0dXJuIG9wZXJhbmQuZGF0ZVRpbWU7CiAgICAgICAgfQoKICAgICAgICAvLy8gPHN1bW1hcnk+CiAgICAgICAgLy8vIENvbnZlcnQgQ2xyIERhdGVUaW1lIHRvIERhdGUKICAgICAgICAvLy8gPC9zdW1tYXJ5PgogICAgICAgIC8vLyA8cGFyYW0gbmFtZT0ib3BlcmFuZCI+RGF0ZVRpbWUgVmFsdWU8L3BhcmFtPgogICAgICAgIC8vLyA8cmV0dXJucz5EYXRlIFZhbHVlIGZyb20gRGF0ZVRpbWU8L3JldHVybnM+CiAgICAgICAgcHVibGljIHN0YXRpYyBpbXBsaWNpdCBvcGVyYXRvciBEYXRlKERhdGVUaW1lIG9wZXJhbmQpCiAgICAgICAgewogICAgICAgICAgICByZXR1cm4gbmV3IERhdGUob3BlcmFuZC5ZZWFyLCBvcGVyYW5kLk1vbnRoLCBvcGVyYW5kLkRheSk7CiAgICAgICAgfQoKICAgICAgICAvLy8gPHN1bW1hcnk+CiAgICAgICAgLy8vIENvbnZlcnQgRGF0ZSB0byBTdHJpbmcKICAgICAgICAvLy8gPC9zdW1tYXJ5PgogICAgICAgIC8vLyA8cmV0dXJucz5zdHJpbmcgdmFsdWUgb2YgRGF0ZTwvcmV0dXJucz4KICAgICAgICBwdWJsaWMgb3ZlcnJpZGUgc3RyaW5nIFRvU3RyaW5nKCkKICAgICAgICB7CiAgICAgICAgICAgIHJldHVybiB0aGlzLmRhdGVUaW1lLlRvU3RyaW5nKCJ5eXl5LU1NLWRkIiwgQ3VsdHVyZUluZm8uSW52YXJpYW50Q3VsdHVyZSk7CiAgICAgICAgfQoKICAgICAgICAvLy8gPHN1bW1hcnk+CiAgICAgICAgLy8vIENvbXBhcmVzIHRoZSB2YWx1ZSBvZiB0aGlzIGluc3RhbmNlIHRvIGEgb2JqZWN0IHZhbHVlCiAgICAgICAgLy8vIGFuZCByZXR1cm5zIGFuIGludGVnZXIgdGhhdCBpbmRpY2F0ZXMgd2hldGhlciB0aGlzIGluc3RhbmNlIGlzIGVhcmxpZXIgdGhhbiwKICAgICAgICAvLy8gdGhlIHNhbWUgYXMsIG9yIGxhdGVyIHRoYW4gdGhlIG9iamVjdCBpZiBpdCBpcyBhIERhdGUuCiAgICAgICAgLy8vIDwvc3VtbWFyeT4KICAgICAgICAvLy8gPHBhcmFtIG5hbWU9Im9iaiI+VGhlIG9iamVjdCB0byBjb21wYXJlIHRvIHRoZSBjdXJyZW50IGluc3RhbmNlPC9wYXJhbT4KICAgICAgICAvLy8gPHJldHVybnM+VmFsdWUgRGVzY3JpcHRpb24gTGVzcyB0aGFuIHplcm8gVGhpcyBpbnN0YW5jZSBpcyBlYXJsaWVyIHRoYW4gdmFsdWUuCiAgICAgICAgLy8vIFplcm8gVGhpcyBpbnN0YW5jZSBpcyB0aGUgc2FtZSBhcyB2YWx1ZS4KICAgICAgICAvLy8gR3JlYXRlciB0aGFuIHplcm8gVGhpcyBpbnN0YW5jZSBpcyBsYXRlciB0aGFuIHZhbHVlLjwvcmV0dXJucz4KICAgICAgICBwdWJsaWMgaW50IENvbXBhcmVUbyhvYmplY3Qgb2JqKQogICAgICAgIHsKICAgICAgICAgICAgaWYgKG9iaiBpcyBEYXRlKQogICAgICAgICAgICB7CiAgICAgICAgICAgICAgICBEYXRlIGRhdGUyID0gKERhdGUpb2JqOwogICAgICAgICAgICAgICAgcmV0dXJuIHRoaXMuQ29tcGFyZVRvKGRhdGUyKTsKICAgICAgICAgICAgfQogICAgICAgICAgICBlbHNlCiAgICAgICAgICAgIHsKICAgICAgICAgICAgICAgIHRocm93IG5ldyBBcmd1bWVudEV4Y2VwdGlvbigiSW52YWxpZCBjb21wYXJlciIpOwogICAgICAgICAgICB9CiAgICAgICAgfQoKICAgICAgICAvLy8gPHN1bW1hcnk+CiAgICAgICAgLy8vIENvbXBhcmVzIHRoZSB2YWx1ZSBvZiB0aGlzIGluc3RhbmNlIHRvIGEgc3BlY2lmaWVkIERhdGUgdmFsdWUKICAgICAgICAvLy8gYW5kIHJldHVybnMgYW4gaW50ZWdlciB0aGF0IGluZGljYXRlcyB3aGV0aGVyIHRoaXMgaW5zdGFuY2UgaXMgZWFybGllciB0aGFuLAogICAgICAgIC8vLyB0aGUgc2FtZSBhcywgb3IgbGF0ZXIgdGhhbiB0aGUgc3BlY2lmaWVkIERhdGUgdmFsdWUuCiAgICAgICAgLy8vIDwvc3VtbWFyeT4KICAgICAgICAvLy8gPHBhcmFtIG5hbWU9Im90aGVyIj5UaGUgb2JqZWN0IHRvIGNvbXBhcmUgdG8gdGhlIGN1cnJlbnQgaW5zdGFuY2U8L3BhcmFtPgogICAgICAgIC8vLyA8cmV0dXJucz5WYWx1ZSBEZXNjcmlwdGlvbiBMZXNzIHRoYW4gemVybyBUaGlzIGluc3RhbmNlIGlzIGVhcmxpZXIgdGhhbiB2YWx1ZS4KICAgICAgICAvLy8gWmVybyBUaGlzIGluc3RhbmNlIGlzIHRoZSBzYW1lIGFzIHZhbHVlLgogICAgICAgIC8vLyBHcmVhdGVyIHRoYW4gemVybyBUaGlzIGluc3RhbmNlIGlzIGxhdGVyIHRoYW4gdmFsdWUuPC9yZXR1cm5zPgogICAgICAgIHB1YmxpYyBpbnQgQ29tcGFyZVRvKERhdGUgb3RoZXIpCiAgICAgICAgewogICAgICAgICAgICByZXR1cm4gdGhpcy5kYXRlVGltZS5Db21wYXJlVG8ob3RoZXIuZGF0ZVRpbWUpOwogICAgICAgIH0KCiAgICAgICAgLy8vIDxzdW1tYXJ5PgogICAgICAgIC8vLyBDb21wYXJlcyB0aGUgdmFsdWUgb2YgdGhpcyBpbnN0YW5jZSB0byBhIHNwZWNpZmllZCBEYXRlIHZhbHVlCiAgICAgICAgLy8vIGFuZCByZXR1cm5zIGFuIGJvb2wgdGhhdCBpbmRpY2F0ZXMgd2hldGhlciB0aGlzIGluc3RhbmNlIGlzIHNhbWUgYXMgdGhlIHNwZWNpZmllZCBEYXRlIHZhbHVlLgogICAgICAgIC8vLyA8L3N1bW1hcnk+CiAgICAgICAgLy8vIDxwYXJhbSBuYW1lPSJvdGhlciI+VGhlIG9iamVjdCB0byBjb21wYXJlIHRvIHRoZSBjdXJyZW50IGluc3RhbmNlPC9wYXJhbT4KICAgICAgICAvLy8gPHJldHVybnM+VHJ1ZSBmb3IgZXF1YWwsIGZhbHNlIGZvciBub24tZXF1YWwuPC9yZXR1cm5zPgogICAgICAgIHB1YmxpYyBib29sIEVxdWFscyhEYXRlIG90aGVyKQogICAgICAgIHsKICAgICAgICAgICAgcmV0dXJuIHRoaXMuZGF0ZVRpbWUuRXF1YWxzKG90aGVyLmRhdGVUaW1lKTsKICAgICAgICB9CgogICAgICAgIC8vLyA8c3VtbWFyeT4KICAgICAgICAvLy8gQ29tcGFyZXMgdGhlIHZhbHVlIG9mIHRoaXMgaW5zdGFuY2UgdG8gYSBzcGVjaWZpZWQgb2JqZWN0IHZhbHVlCiAgICAgICAgLy8vIGFuZCByZXR1cm5zIGFuIGJvb2wgdGhhdCBpbmRpY2F0ZXMgd2hldGhlciB0aGUgc3BlY2lmaWVkIG9iamVjdCBpcyA8c2VlIGNyZWY9Ik1pY3Jvc29mdC5PRGF0YS5FZG0uRGF0ZSIvPgogICAgICAgIC8vLyBhbmQgdGhpcyBpbnN0YW5jZSBpcyBzYW1lIGFzIHRoZSBzcGVjaWZpZWQgPHNlZSBjcmVmPSJNaWNyb3NvZnQuT0RhdGEuRWRtLkRhdGUiLz4gdmFsdWUuCiAgICAgICAgLy8vIDwvc3VtbWFyeT4KICAgICAgICAvLy8gPHBhcmFtIG5hbWU9Im9iaiI+VGhlIG9iamVjdCB0byBjb21wYXJlIHRvIHRoZSBjdXJyZW50IGluc3RhbmNlPC9wYXJhbT4KICAgICAgICAvLy8gPHJldHVybnM+VHJ1ZSBmb3IgZXF1YWwsIGZhbHNlIGZvciBub24tZXF1YWwuPC9yZXR1cm5zPgogICAgICAgIHB1YmxpYyBvdmVycmlkZSBib29sIEVxdWFscyhvYmplY3Qgb2JqKQogICAgICAgIHsKICAgICAgICAgICAgaWYgKG9iaiA9PSBudWxsIHx8ICEob2JqIGlzIERhdGUpKQogICAgICAgICAgICB7CiAgICAgICAgICAgICAgICByZXR1cm4gZmFsc2U7CiAgICAgICAgICAgIH0KCiAgICAgICAgICAgIHJldHVybiB0aGlzLmRhdGVUaW1lLkVxdWFscygoKERhdGUpb2JqKS5kYXRlVGltZSk7CiAgICAgICAgfQoKICAgICAgICAvLy8gPHN1bW1hcnk+CiAgICAgICAgLy8vIFJldHVybnMgdGhlIGhhc2ggY29kZSBmb3IgdGhpcyBpbnN0YW5jZS4KICAgICAgICAvLy8gPC9zdW1tYXJ5PgogICAgICAgIC8vLyA8cmV0dXJucz5BIDMyLWJpdCBzaWduZWQgaW50ZWdlciBoYXNoIGNvZGUuPC9yZXR1cm5zPgogICAgICAgIHB1YmxpYyBvdmVycmlkZSBpbnQgR2V0SGFzaENvZGUoKQogICAgICAgIHsKICAgICAgICAgICAgcmV0dXJuIHRoaXMuZGF0ZVRpbWUuR2V0SGFzaENvZGUoKTsKICAgICAgICB9CgogICAgICAgIC8vLyA8c3VtbWFyeT4KICAgICAgICAvLy8gQ29udmVydCBEYXRlIHRvIENsciBEYXRlT25seQogICAgICAgIC8vLyA8L3N1bW1hcnk+CiAgICAgICAgLy8vIDxwYXJhbSBuYW1lPSJvcGVyYW5kIj5EYXRlIFZhbHVlPC9wYXJhbT4KICAgICAgICAvLy8gPHJldHVybnM+RGF0ZVRpbWUgVmFsdWUgd2hpY2ggcmVwcmVzZW50IHRoZSBEYXRlPC9yZXR1cm5zPgogICAgICAgIHB1YmxpYyBzdGF0aWMgaW1wbGljaXQgb3BlcmF0b3IgRGF0ZU9ubHkoRGF0ZSBvcGVyYW5kKQogICAgICAgIHsKICAgICAgICAgICAgcmV0dXJuIERhdGVPbmx5LkZyb21EYXRlVGltZShvcGVyYW5kLmRhdGVUaW1lKTsKICAgICAgICB9CgogICAgICAgIC8vLyA8c3VtbWFyeT4KICAgICAgICAvLy8gQ29udmVydCBDbHIgRGF0ZU9ubHkgdG8gRGF0ZQogICAgICAgIC8vLyA8L3N1bW1hcnk+CiAgICAgICAgLy8vIDxwYXJhbSBuYW1lPSJvcGVyYW5kIj5EYXRlT25seSBWYWx1ZTwvcGFyYW0+CiAgICAgICAgLy8vIDxyZXR1cm5zPkRhdGUgVmFsdWUgZnJvbSBEYXRlT25seTwvcmV0dXJucz4KICAgICAgICBwdWJsaWMgc3RhdGljIGltcGxpY2l0IG9wZXJhdG9yIERhdGUoRGF0ZU9ubHkgb3BlcmFuZCkKICAgICAgICB7CiAgICAgICAgICAgIHJldHVybiBuZXcgRGF0ZShvcGVyYW5kLlllYXIsIG9wZXJhbmQuTW9udGgsIG9wZXJhbmQuRGF5KTsKICAgICAgICB9CiAgICB9Cn0K).

This was meant to verify whether there's an actual overhead of casting `DateOnly` to `Edm.Date` and that we can avoid that perf overhead by handling `DateOnly` directly